sns_ik_lib/test/sns_ik_math_utils_test.cpp

```diff
@@ -410,6 +442,61 @@ TEST(sns_ik_math_utils, pseudoInverse_damped_test)
 
 /*************************************************************************************************/
 
+TEST(sns_ik_math_utils, solveLinearSystem_fullRank_test)
+{
+  // test parameters
+  double tol = 1e-10;  // tolerance for matrix equality check
+  int nTest = 25;
+  int s1 = 84658;  // seed for the integer RNG
+  int s2 = 63541;  // seed for the floating point RNG
+  double low = -2.0;  double upp = 2.0;  // bounds on values in the A matrix
+  // test setup
+  Eigen::MatrixXd A, b;  // test linear system
+  Eigen::MatrixXd x;  // result
+  int rank; // rank of A, computed by linear solver
+  double res;  // residual error in the solution
+  for (int iTest = 0; iTest < nTest; iTest++) {
+    int n = sns_ik::rng_util::getRngInt(s1, 3, 9);
+    int m = sns_ik::rng_util::getRngInt(0, n, n + 5);
+    int p = sns_ik::rng_util::getRngInt(0, 1, 3);
+    A = sns_ik::rng_util::getRngMatrixXd(s2, n, m, low, upp);
+    b = sns_ik::rng_util::getRngMatrixXd(s2, n, p, low, upp);
+    ASSERT_TRUE(sns_ik::solveLinearSystem(A, b, &x, &rank, &res));
+    checkLinearSolve(A, b, x, rank, res, tol);
+    s1 = s2 = 0; // let the RNG automatically increment after the first iteration
```

Comment:
How is it that the RNG is incrementing?

Reply:
The normal operation of a random number generator is to give the next number in the sequence on each call. This is what happens whenever you pass seed = 0 in my functions. If you pass a non-zero seed, then I override the next number in the sequence by setting the seed. This allows a repeatable result. In this code I try to only reset the RNG seed at the start of each individual test, allowing it to use the normal sequence of numbers that follow from that seed.